src/com/simulator/controller/SmartHomeSimulatorController.java

```diff
@@ -300,24 +303,100 @@ private void setHouseView() {
         }
     }
 
-        //Ended here
     @FXML
-    private void addRoomToZoneFunction(){
+    private void addRoomToSelectedRoomsPanel(){
+        Object selectedItem = this.allRoomsListViewZone.getSelectionModel().getSelectedItem();
+        this.selectedRoomsListForZone.getItems().add((String) selectedItem);
+        this.allRoomsListViewZone.getItems().remove(selectedItem);
+    }
 
+    @FXML
+    private void removeRoomFromSelectedRoomsPanel(){
+        Object selectedItem = this.selectedRoomsListForZone.getSelectionModel().getSelectedItem();
+        this.selectedRoomsListForZone.getItems().remove((String) selectedItem);
+        this.allRoomsListViewZone.getItems().add(selectedItem);
     }
     @FXML
-    private void removeRoomFromZoneFunction(){
+    private void addPresetTemperatureToZone(){
+        if(temperatureTextField.getText() ==null || temperatureTextField.getText().trim().isEmpty()||!temperatureTextField.getText().matches("[0-9]+")){
+            Logger.getInstance().outputToConsole("Please make sure a valid temperature is entered");
+            return;
+        }
+        int tempPreset = Integer.valueOf(temperatureTextField.getText());
+        if(listZones.getSelectionModel().isEmpty()){
+            Logger.getInstance().outputToConsole("Please make sure you have selected a zone");
+            return;
+        }
+        String selectedZone = (String)listZones.getSelectionModel().getSelectedItem();
+        if(temperatureComboBox.getSelectionModel().isEmpty()){
+            Logger.getInstance().outputToConsole("Please make sure you have selected a period of time");
+            return;
+        }
+        String selectedPeriodOfTheDay = (String)temperatureComboBox.getSelectionModel().getSelectedItem();
 
+        //Set the new temperature, unless one has already been overwritten for a specific room
+        heatingModule.setTempForZone(selectedZone, selectedPeriodOfTheDay, tempPreset);
+        System.out.println("Came out the other side brother");
     }
+
     @FXML
-    private void setRoomsZonePanel(){
-        ArrayList<String> roomNameList = (ArrayList<String>) House.getInstance().getRoomsNameList();
+    private void createZoneButton(){
+        //Check to see if the user selected some entries
+        if(selectedRoomsListForZone.getItems().size()==0){
+            Logger.getInstance().outputToConsole("Please select some entries for the zone");
+            return;
+        }
 
-        for (String room: roomNameList){
-            allRoomsListViewZone.getItems().add(room);
+        //check to see if the string already exists or if an invalid length string was entered
+        if(zoneName.getText().length()<1||!heatingModule.checkIfValidZoneName(zoneName.getText())){
```

Comment:
This may not be necessary but it is good practice to have a space around your operators 
```suggestion
        if(zoneName.getText().length() < 1 || !heatingModule.checkIfValidZoneName(zoneName.getText())){
```

is more readable I saw a couple more cases where you are missing spaces around the operator but this is not necessary just good practice so if you want to just do a quick review and add some spaces doesn't have to be perfect